### PR TITLE
Remove hardcoded user profile fields

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

_Note: This is the replacement PR for #13996 it was just easier to create a new branch._

## Related Tickets & Documents

Profile generalization

## QA Instructions, Screenshots, Recordings

This is a pretty far-reaching change, so the more you can manually test the better. Some ideas:

* Creating and viewing articles
* Updating various profile settings
* Looking at different user profiles
* Creating listings

### UI accessibility concerns?

None, this PR primarily changes implementation details.

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] I will share this change internally with the appropriate teams

### CODE REVIEW

File: spec/requests/user/user_settings_spec.rb [hunk 1]
```diff
@@ -238,14 +238,9 @@
   describe "PUT /update/:id" do
     before { sign_in user }
 
-    it "updates summary" do
```
Comment (citizen428) [inline spec/requests/user/user_settings_spec.rb:241]: The profile tab no longer posts to this controller so we can remove the spec from here. That it still worked at all was more of side-effect of the delegation than an actual feature.

**Comment by citizen428** [comment]
The test failures are interesting:

* `spec/requests/profile_preview_cards_spec.rb`: is always green for me locally but consistently fails on Travis.
* The e2e seed scripts raises an error on Travis for something I already fixed but work fine for me locally.

**Comment by citizen428** [comment]
While there's still one failing e2e test I want to open this PR up for review/testing now.

File: app/models/profile_field.rb [hunk 2]
```diff
@@ -13,8 +13,7 @@ class ProfileField < ApplicationRecord
 
   enum display_area: {
     header: 0,
-    left_sidebar: 1,
-    settings_only: 2
```
Comment (rhymes) [inline app/models/profile_field.rb:17]: what happens to rows that have this value in the DB?

File: app/views/users/_meta.html.erb [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 
 <meta property="og:type" content="website" />
 <meta property="og:url" content="<%= user_url(@user) %>" />
 <meta property="og:title" content="<%= @user.name %> — <%= community_name %> Profile" />
 <meta property="og:image" content="<%= user_social_image_url(@user) %>">
-<meta property="og:description" content="<%= @user.profile.summary %>" />
+<meta property="og:description" content="<%= @user.tag_line %>" />
 <meta property="og:site_name" content="<%= community_name %>" />
-
+    <meta name="twitter:card" content="summary_large_image">
```
Comment (rhymes) [inline app/views/users/_meta.html.erb:11]: isn't this the same as the following line?

File: app/models/profile_field.rb (hunk 2 above)
Comment (citizen428) [reply app/models/profile_field.rb:17]: They were already removed with a DUS here: https://github.com/forem/forem/pull/14206

File: app/views/users/_meta.html.erb (hunk 3 above)
Comment (citizen428) [reply app/views/users/_meta.html.erb:11]: Yes, but the indentation is different. Maybe a merge artifact, thanks!

**Review by Ridhwana — APPROVED** [review]
This looks good to me.

**Review by rhymes — APPROVED** [review]
LGTM!

**Pull request merged**